backend/src/govsg/services/govsg.service.ts

```diff
@@ -111,11 +111,18 @@ export function uploadCompleteOnPreview({
   campaignId: number
 }): (data: CSVParams[]) => Promise<void> {
   return async function (data: CSVParams[]): Promise<void> {
+    const paramsWithoutPasscode = template.params!.filter(
```

Comment:
Do we need to address this warning?

Reply:
By warning do you mean `!`?

The value has to be valid or we don't want to continue ahead.

`!` is not my preference but I would deem this out-of-scope. We should really have caught this during code review but we weren't involved back then.